components/pluralrules/src/operands.rs

```diff
@@ -0,0 +1,275 @@
+            fraction_digits,
+        ) = if let Some(sep_idx) = abs_str.find('.') {
+            let int_str = &abs_str[..sep_idx];
+            let dec_str = &abs_str[(sep_idx + 1)..];
+
+            let integer_digits = u64::from_str(&int_str)?;
+
+            let dec_str_no_zeros = dec_str.trim_end_matches('0');
+
+            let num_fraction_digits0 = dec_str.len() as usize;
+            let num_fraction_digits = dec_str_no_zeros.len() as usize;
+
+            let fraction_digits0 = u64::from_str(&dec_str)?;
+            let fraction_digits =
+                if num_fraction_digits == 0 || num_fraction_digits == num_fraction_digits0 {
+                    fraction_digits0
+                } else {
+                    u64::from_str(&dec_str_no_zeros)?
+                };
+
+            let fraction: f64 = (fraction_digits as f64) / 10_f64.powi(num_fraction_digits as i32);
+
+            (
+                integer_digits as f64 + fraction,
+                integer_digits,
+                num_fraction_digits0,
+                num_fraction_digits,
+                fraction_digits0,
+                fraction_digits,
+            )
+        } else {
+            let integer_digits = u64::from_str(&abs_str)?;
+            (integer_digits as f64, integer_digits, 0, 0, 0, 0)
+        };
+
+        Ok(PluralOperands {
+            n: absolute_value,
+            i: integer_digits,
+            v: num_fraction_digits0,
+            w: num_fraction_digits,
+            f: fraction_digits0,
+            t: fraction_digits,
+        })
+    }
+}
+
+macro_rules! impl_integer_type {
+    ($ty:ident) => {
+        impl From<$ty> for PluralOperands {
+            fn from(input: $ty) -> Self {
+                PluralOperands {
+                    n: input as f64,
+                    i: input as u64,
+                    v: 0,
+                    w: 0,
+                    f: 0,
+                    t: 0,
+                }
+            }
+        }
+    };
+    ($($ty:ident)+) => {
+        $(impl_integer_type!($ty);)+
+    };
+}
+
+macro_rules! impl_signed_integer_type {
+    ($ty:ident) => {
+        impl TryFrom<$ty> for PluralOperands {
+            type Error = OperandsError;
+            fn try_from(input: $ty) -> Result<Self, Self::Error> {
+                let x = input.checked_abs().ok_or(OperandsError::Invalid)?;
+                Ok(PluralOperands {
+                    n: x as f64,
+                    i: x as u64,
+                    v: 0,
+                    w: 0,
+                    f: 0,
+                    t: 0,
+                })
+            }
+        }
+    };
+    ($($ty:ident)+) => {
+        $(impl_signed_integer_type!($ty);)+
+    };
+}
+
+// This algorithms allows us to use the Rust's decimal to float parser
+// using on-stack buffer.
+//
+// Once we start relying on Ryu, we should consider swapping it to use that.
+fn get_dec_str_from_f64(absolute_value: f64, buffer: &mut [u8]) -> Result<&str, IOError> {
```

Comment:
I've said this before, and I'll say it again: PluralRules *cannot* be correctly implemented from a floating point type, because floating point types are not capable of carrying trailing zeros, which are crucial to plural rules correctness.  I claim that if a programmer is passing f64s into PluralRules, then they *necessarily* have a bug in their code.  There is *no way* to use an f64 plural operand selector correctly in the absence of ways to specify minimum fraction digits, which is *exactly why* we added FixedDecimal.

I would strongly prefer if you simply remove this method.  People should be using FixedDecimal as input to PluralRules if they care about fractional digits.

Reply:
Could we at least evaluate the perf impact between this vs. via `FixedDecimal` before we remove it? I'm ok marking it for removal in the doc and opening an issue to evaluate performance of the `FixedDecimal` backed approach.

Reply:
I don't think it's useful or productive to compare performance against an incorrect algorithm.

Reply:
Do you plan to allow people to parse `f64` into `FixedDecimal`?

Reply:
I haven't made a PR yet, but I anticipate adding a `f64` to `FixedDecimal` that, for example, takes a second argument, the number of fraction digits.

I think it isn't possible to write `impl From<f64> for FixedDecimal` because of the trailing zeros problem.  There is no reasonable default for trailing zeros that doesn't hold the potential for incorrect behavior.

Reply:
Ah, my thinking was that you may allow `From<f64>` for `FixedDecimal` and then adjust magnitude.

I recognize that I know very little about the space, but I can totally imagine someone trying to construct `FixedDecimal` from `f64` and expecting it to work (with an assumption that `2.30` and `2.3` will be treated the same way)

Reply:
I removed floats from this PR in https://github.com/unicode-org/icu4x/pull/119/commits/f589e627049d3f55092a5b8cfc315aefe56ec45d and filed https://github.com/unicode-org/icu4x/issues/191 to discuss further.